workflow/controller/operator.go

```diff
@@ -2624,3 +2646,60 @@ func (woc *wfOperationCtx) includeScriptOutput(nodeName, boundaryID string) (boo
 	}
 	return false, nil
 }
+
+func (woc *wfOperationCtx) getTopLevelWorkflowTemplate() (*wfv1.WorkflowSpec, error) {
```

Comment:
Don't use the term `TopLevel` here. I would call this
```suggestion
func (woc *wfOperationCtx) fetchWorkflowSpec() (*wfv1.WorkflowSpec, error) {
```

Reply:
Furthermore, I think this should be a utility function, not a member of `wfOperationCtx`. You can pass in `woc.wf.Spec.WorkflowTemplateRef` as a parameter

Reply:
I thought too. But function needs to access the informer from wfOperationCtx. I thought instead of passing all references. This function is going used only in WorkflowTemplate usecase. it would be better to have it simple and clean.